docs/usage_guide/optimization_guide/index.rst

```diff
@@ -0,0 +1,58 @@
+Optimization Guide
```

Comment:
I would change this name to an advanced optimization guide, and maybe provide a link to the `modin/core/dataframe/pandas/dataframe/dataframe.py` file where the algebra is located with a quick high-level description in case a less advanced user looks at this.

Reply:
Not sure this requires renaming and putting the link.

Reply:
It would likely be better to rename this name to `Best Practices`.

Reply:
I tend to leave the title as it is, this page is more about a guide, rather than just listing best practices for Modin, since we're giving a brief about partitioning and all the related stuff.

Reply:
I tend to agree with @naren-ponder on this one, since it is more for advanced users. I do not feel strongly either way.

Reply:
What is the difference between an advanced optimization guide and just an optimization guide? We are going to provide both intuitive tips and tricky ones. I think everything of those should lay on just opmization_guide page.

Reply:
Replaced to optimization notes. Also, I think it should be divided into separate sections: pandas_on_ray/dask, omnisci_on_native.